src/core/lib/slice/slice_internal.h

```diff
@@ -327,4 +327,32 @@ inline grpc_slice grpc_slice_from_static_string_internal(const char* s) {
   return grpc_slice_from_static_buffer_internal(s, strlen(s));
 }
 
+/* Helper method for grpc_slice_buffer_move_into_internal only - used when
+   dst is empty. Only grpc_slice_buffer_move_into_internal() should be used.
+   Think of this as a static method, not part of the internal API. */
+void _helper_grpc_slice_buffer_move_into_empty_buf(grpc_slice_buffer* src,
+                                                   grpc_slice_buffer* dst);
+/* Helper method for grpc_slice_buffer_move_into_internal only - used when both
+   src and dst are non-empty. While this method still executes correctly
+   if directly called (including when src and/or dst are empty), only
+   grpc_slice_buffer_move_into_internal() should be used. Think of this as a
+   static method, not part of the internal API. */
+void _helper_grpc_slice_buffer_move_into_buf(grpc_slice_buffer* src,
+                                             grpc_slice_buffer* dst);
+/* Move all of the elements of src into dst. This is part of the internal API.
+   It makes no assumptions on either src or dst. */
```

Comment:
I think this explanation is too vague from the perspective of someone reading this and trying to understand the API.  What does it mean to "make no assumptions"?

I think it would be better to state the preconditions (e.g., both `src` and `dst` must be initialized) and postconditions (e.g., any pre-existing slices in `dst` will be unreffed).